lib/fleiss/backend/active_record/concern.rb

```diff
@@ -26,12 +37,14 @@ def wrap_perform(&block)
 
           # @return [ActiveRecord::Relation] pending scope
           def pending(now = Time.zone.now)
-            not_finished
-              .not_expired(now)
-              .not_started
-              .where(arel_table[:scheduled_at].lteq(now))
-              .order(priority: :desc)
-              .order(scheduled_at: :asc)
+            pending = not_finished
+                      .not_expired(now)
+                      .not_started
+                      .where(arel_table[:scheduled_at].lteq(now))
+                      .order(priority: :desc)
+                      .order(scheduled_at: :asc)
+
+            pending.or(lock_expired(now, DEFAULT_LOCK_TTL))
```

Comment:
that's not right. if something is e.g. finished or expired, you don't want to include it!!
it should be:

```rb
not_finished
  .not_exipred
  .where(arel_table[:scheduled_at].lteq(now))
  .merge(not_started(now).or(lock_expired(now))
  .order(priority: :desc, scheduled_at: :asc)
```

Reply:
:facepalm: